sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1108,6 +1109,18 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 		if (!name)
 			return -ENOMEM;
 
+		for (index = 0; index < *link_index; index++) {
+			/*
+			 * Append codec dai name to dai link name if the dai link name is
+			 * duplicated.
+			 */
+			if (!strcmp(name, dai_links[index].name)) {
+				devm_kfree(dev, name);
+				name = devm_kasprintf(dev, GFP_KERNEL, "%s-%s",
+						      dai_links[index].name,
+						      codec_info_list[codec_index].dai_name);
```

Comment:
do we really want to do this? This would require us to have codec-specific stuff in the topology, no?

Reply:
> do we really want to do this? This would require us to have codec-specific stuff in the topology, no?

Maybe we can rename `codec_type` to `dai_type` in `sof_sdw_codec_info{}` and construct dai name with dai_type?
For example, `SDW0-Playback-amp`, ` SDW0-Playback-jack`